.github/workflows/add_release_assets.yml

```diff
@@ -17,10 +22,55 @@ jobs:
       - uses: actions/checkout@08c6903cd8c0fde910a37f88322edcfb5dd907a8 # v5.0.0
         with:
           persist-credentials: false
-      - run: echo "tag_name=$(gh release list --repo ${{ github.repository }} --json 'isDraft,tagName' --jq '.[] | select(.isDraft) | .tagName')" >> $GITHUB_OUTPUT
+      - name: Get tag name
+        id: get-tag
+        run: |
+          # Use manual input if provided, otherwise auto-detect from draft releases
+          if [ -n "${{ inputs.tag_name }}" ]; then
+            echo "Using manual tag name: ${{ inputs.tag_name }}"
+            echo "tag_name=${{ inputs.tag_name }}" >> $GITHUB_OUTPUT
+          else
+            echo "Auto-detecting tag from draft releases..."
+            
+            # Try to get draft releases with retry logic
+            # Note: gh CLI automatically uses GITHUB_TOKEN from environment
+            MAX_RETRIES=3
+            RETRY_COUNT=0
+            TAG_NAME=""
+            
+            while [ $RETRY_COUNT -lt $MAX_RETRIES ] && [ -z "$TAG_NAME" ]; do
+              if [ $RETRY_COUNT -gt 0 ]; then
+                echo "Retry attempt $RETRY_COUNT of $MAX_RETRIES..."
+                sleep 2
+              fi
+              
+              # Get draft releases and show raw output for debugging
+              echo "Fetching draft releases from ${{ github.repository }}..."
+              RELEASES_JSON=$(gh release list --repo ${{ github.repository }} --json 'isDraft,tagName' 2>&1) || true
+              echo "Raw response: $RELEASES_JSON"
+              
+              # Extract the first draft release tag
+              TAG_NAME=$(echo "$RELEASES_JSON" | jq -r '.[] | select(.isDraft == true) | .tagName' 2>/dev/null | head -1)
+              
+              RETRY_COUNT=$((RETRY_COUNT + 1))
+            done
```

Comment:
Simplify this all back down, take out the retries, remove some of the output. This is way too much code for what it's doing.